### PR TITLE
Extract constants in node patterns to node patterns

### PR DESCRIPTION
**Pull request by pirj**
* [x] works on our lower-bound RuboCop 0.87.0

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [-] Added tests.
* [-] Updated documentation.
* [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: lib/rubocop/rspec/language/node_pattern.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ module Language
       module NodePattern
         extend RuboCop::NodePattern::Macros
 
+        def_node_matcher :rspec?, '{(const {nil? cbase} :RSpec) nil?}'
```
Comment (pirj) [inline lib/rubocop/rspec/language/node_pattern.rb:10]: For some reason, I had an idea that this is impossible.

But is it good?
Using `#{RSPEC}` in several node patterns mean we'll compiling this part into Ruby code each time. For `#rspec`, we're compiling once and reusing the already compiled code across other node matchers that use it.

**Comment by pirj** [comment]
Ping @bquorning @Darhazer

File: lib/rubocop/rspec/language/node_pattern.rb (hunk 1 above)
Comment (Darhazer) [reply lib/rubocop/rspec/language/node_pattern.rb:10]: > For some reason, I had an idea that this is impossible.

Neither did I :joy:

**Comment by pirj** [comment]
Ping @bquorning

**Pull request merged**